cmd/arcstat.in

```diff
@@ -169,6 +170,74 @@ cols = {
     "zactive":    [7, 1000, "zfetch prefetches active per second"],
 }
 
+# ARC structural breakdown from arc_summary
+structfields = {
+    "cp":       ["compressed", "Compressed"],
+    "oh":       ["overhead", "Overhead"],
+    "bn":       ["bonus", "Bonus"],
+    "dn":       ["dnode", "Dnode"],
+    "db":       ["dbuf", "Dbuf"],
+    "hdr":      ["hdr", "Header"],
+    "l2h":      ["l2_hdr", "L2 header"],
+    "abd":      ["abd_chunk_waste", "ABD chunk waste"],
+}
+structstats = {                                     # size stats
+    "percent":  "size",                             # show percentage of this value
+    "sz":       ["_size", "size"],
+}
+
+# ARC types breakdown from arc_summary
+typefields = {
+    "data":     ["data", "ARC data"],
+    "meta":     ["metadata", "ARC metadata"],
+}
+typestats = {                                       # size stats
+    "percent":  "cachessz",                         # show percentage of this value
+    "sz":       ["_size", "size"],
+}
+
+# ARC states breakdown from arc_summary
+statefields = {
+    "ano":      ["anon", "Anonymous"],
+    "mfu":      ["mfu", "MFU"],
+    "mru":      ["mru", "MRU"],
+    "unc":      ["uncached", "Uncached"],
+}
+statestats = {                                      # size stats
+    "percent":  "cachessz",                         # show percentage of this value
+    "sz":       ["_size", "size"],
+    "data":     ["_data", "data size"],
+    "meta":     ["_metadata", "metadata size"],
+    "eda":      ["_evictable_data", "evictable data size"],
+    "eme":      ["_evictable_metadata", "evictable metadata size"],
+}
+gstatestats = {                                     # ghost stats
+    "fields":   ["mfu", "mru"],                     # only applicable to these fields
+    "gsz":      ["_ghost_size", "ghost size"],
+    "gda":      ["_ghost_data", "ghost data size"],
+    "gme":      ["_ghost_metadata", "ghost metadata size"],
+}
+
+# fields and stats
+fieldstats = [
+    [structfields, structstats],
+    [typefields, typestats],
+    [statefields, statestats, gstatestats],
+]
+for fs in fieldstats:
+    (fields, stats) = fs[0], fs[1:]
+    for field, fieldval in fields.items():
+        for group in stats:
+            for stat, statval in group.items():
+                if stat in ["fields", "percent"] or \
+                    ("fields" in group and field not in group["fields"]):
+                    continue
+                colname = field + stat
+                coldesc = fieldval[1] + " " + statval[1]
+                cols[colname] = [len(colname), 1024, coldesc]
```

Comment:
The column width should account not only name length, but also value length, which for 1024-base may be 5, while this code I suppose will produce 4 for `cpsz`, etc.